src/main/java/edu/harvard/iq/dataverse/batch/jobs/importer/filesystem/FileRecordWriter.java

```diff
@@ -166,6 +167,13 @@ public void writeItems(List list) {
                     jobContext.setExitStatus("FAILED");
                     return;
                 }
+                DatasetLock dcmLock = dataset.getLockFor(DatasetLock.Reason.DcmUpload);
```

Comment:
Its kind of confusing that a writeItems method is removing a lock and updating the dataset version. is there a reason this doesn't live in the close method?

Reply:
Ahh ok. Thanks for the clarification!